File: src/main/java/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtClassNameMixin.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2017 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins;

import com.intellij.psi.PsiClass;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface AtClassNameMixin extends PsiElement {

    @NotNull
    PsiElement getClassNameElement();

    @Nullable
    PsiClass getClassNameValue();

    @NotNull
    String getClassNameText();

    void setClassName(@NotNull String className);
}

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtClassNameMixin.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2019 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins

import com.demonwav.mcdev.platform.mcp.at.psi.AtElement
import com.intellij.psi.PsiClass
import com.intellij.psi.PsiElement

interface AtClassNameMixin : AtElement {

    val classNameElement: PsiElement
    val classNameValue: PsiClass?
    val classNameText: String

    fun setClassName(className: String)
}
